main-command/src/main/java/sbt/internal/BootServerSocket.java

```diff
@@ -281,13 +281,23 @@ public OutputStream outputStream() {
         }
       };
 
+  private static String getUnixParentDirectory(String path) {
+    for (int i = path.length() - 2; i >= 0; i--) {
+      if (path.charAt(i) == '\\' || path.charAt(i) == '/') {
+        return path.substring(0, i + 1);
+      }
+    }
+    return "/";
+  }
```

Comment:
Is this required as opposed to using NIO Path?

Reply:
I have updated the code in order to use NIO Path.getParent()